miden-lib/src/notes/scripts.rs

```diff
@@ -1,25 +1,41 @@
-use miden_objects::{notes::NoteScript, utils::Deserializable, vm::Program};
+use miden_objects::{
+    notes::NoteScript,
+    utils::{sync::LazyLock, Deserializable},
+    vm::Program,
+};
 
-/// Returns a P2ID (Pay-to-ID) note script.
-pub fn p2id() -> NoteScript {
+// Initialize the P2ID note script only once
+static P2ID_SCRIPT: LazyLock<NoteScript> = LazyLock::new(|| {
     let bytes = include_bytes!(concat!(env!("OUT_DIR"), "/assets/note_scripts/P2ID.masb"));
     let program = Program::read_from_bytes(bytes).expect("Shipped P2ID script is well-formed");
-
     NoteScript::new(program)
-}
+});
 
-/// Returns a P2IDR (Pay-to-ID with recall) note script.
-pub fn p2idr() -> NoteScript {
+// Initialize the P2IDR note script only once
+static P2IDR_SCRIPT: LazyLock<NoteScript> = LazyLock::new(|| {
     let bytes = include_bytes!(concat!(env!("OUT_DIR"), "/assets/note_scripts/P2IDR.masb"));
     let program = Program::read_from_bytes(bytes).expect("Shipped P2IDR script is well-formed");
-
     NoteScript::new(program)
-}
+});
 
-/// Returns a SWAP (Pay-to-ID with recall) note script.
-pub fn swap() -> NoteScript {
+// Initialize the SWAP note script only once
+static SWAP_SCRIPT: LazyLock<NoteScript> = LazyLock::new(|| {
     let bytes = include_bytes!(concat!(env!("OUT_DIR"), "/assets/note_scripts/SWAP.masb"));
     let program = Program::read_from_bytes(bytes).expect("Shipped SWAP script is well-formed");
-
     NoteScript::new(program)
+});
+
+/// Returns the P2ID (Pay-to-ID) note script.
+pub fn p2id() -> &'static NoteScript {
+    &P2ID_SCRIPT
+}
```

Comment:
Would it make sense to clone the script and return `NoteScript` from here since we end up cloning it downstream anyways? For example:
```rust
pub fn p2id() -> NoteScript {
    P2ID_SCRIPT.clone()
}
```
Also, cloning of `NoteScript` is a pretty light-weight operation as it consists of a single `u32` and an `Arc` pointer.

Reply:
Yeah, I was returning a reference so that things like the the note script hash would be easily accessible with no cloning but I see now that there should be very negligible cost to cloning.